pkg/util/cluster/cluster.go

```diff
@@ -169,9 +160,9 @@ func (c *Cluster) DeleteApp(ctx context.Context) error {
 }
 
 func (c *Cluster) Create(ctx context.Context, vnetResourceGroup, clusterName string, osClusterVersion string) error {
-	clusterGet, err := c.openshiftclustersv20230904.Get(ctx, vnetResourceGroup, clusterName)
+	clusterGet, err := c.openshiftclusters.Get(ctx, vnetResourceGroup, clusterName)
```

Comment:
I'm sure you've thought of this, but for the time being, does it make sense to just wrap this into an `if isDevelopment` or similar block, and have prod e2es use the old client? It would make for cleanup later, but not sure how else we'd merge this before the v20240812preview API is available everywhere.

This has the downside of e2e testing the new API in dev but the old API in prod, which might be useful for smoke testing the new API, but would create a slight difference between the e2e runs depending on environment.

Reply:
Team Loki talked about this in our sync PR review session, this proposed solution is somewhere on the spectrum between the current implementation in this PR, and a long-term idea we have to make the API version used here configurable (e.g. allow us to run E2E against multiple API versions), something that may become more desirable for us as we progress further along the CI/CD initiative. 

I would like to implement this proposed solution in the near-term, but I think it would require some thought into re-usability for that potential later step, and thus I think it makes the most sense in a follow-up PR. 

For now, I'm thinking the steps are:
- This PR: downgrade the single API version used to our latest routeable API (20230904)
- Follow-up PR: implement some sort of determination between the latest routeable and in-progress APIs, e.g. through an abstracted client implementation that accepts requests and provides responses with the internal clusterdoc representation, and converts to the specific external representation before actually amking the API call (which is how pkg/util/cluster operates today anyways)
- Future effort, depending on need: allow pkg/util/cluster to be instantiated with a specific API version, implement a configurable parameter in both PR and prod E2E to specify the version to use (default being the behavior above)